iothub/device/src/Authentication/IotHubConnectionStringBuilder.cs

```diff
@@ -65,40 +56,40 @@ public IotHubConnectionStringBuilder(string iotHubConnectionString)
 
             // We'll parse the connection string and use that to build an auth method
             ExtractPropertiesFromConnectionString(iotHubConnectionString);
-            AuthenticationMethod = AuthenticationMethodFactory.GetAuthenticationMethod(this);
+            AuthenticationMethod = AuthenticationMethodFactory.GetAuthenticationMethodFromConnectionString(this);
 
             Validate();
         }
 
         /// <summary>
         /// Gets or sets the value of the fully-qualified DNS hostname of the IoT hub service.
         /// </summary>
-        public string HostName { get; internal set; }
+        public string HostName { get; private set; }
 
         /// <summary>
         /// Gets the optional name of the gateway to connect to
         /// </summary>
-        public string GatewayHostName { get; internal set; }
+        public string GatewayHostName { get; private set; }
 
         /// <summary>
         /// Gets the device identifier of the device connecting to the service.
         /// </summary>
-        public string DeviceId { get; internal set; }
```

Comment:
This and below public properties - these can be set through a custom IAuthenticationMethod implementation through the `Populate` method. So, making them public.

The constructor still calls validate, so the properties would get validated there.

```csharp
AuthenticationMethod.Populate(this);
Validate();
```

Reply:
Why not internal?

Reply:
The Populate method class is on the public IAUthenticationMethod interface that anyone can implement.

Reply:
Ah